SessionSnodeKit/Storage+SnodeAPI.swift

```diff
@@ -110,7 +110,6 @@ extension Storage {
         if now >= expirationDate {
             Storage.writeSync { transaction in
                 self.removeLastMessageHashInfo(for: snode, namespace: namespace, associatedWith: publicKey, using: transaction)
-                self.setReceivedMessages(to: Set(), for: publicKey, using: transaction)
```

Comment:
Was this the reason messages were getting reprocessed?

Reply:
I guess this is. It can happen when there is a snode's last hash out of date. It will remove all received message hashes for all snodes, because the received message hashes are associated with session id, but the last hash info is associated with snode. But I'd say it is a corner case, it shouldn't affect all users.